anta/tests/routing/bgp.py

```diff
@@ -1226,3 +1226,87 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"Following BGP peers are not configured or hold and keep-alive timers are not correct:\n{failures}")
+
+
+class VerifyBGPPeerDropStats(AntaTest):
+    """Verifies BGP NLRI drop statistics for the provided BGP IPv4 peer(s), checking for any non-zero values in various drop categories.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the BGP peer's drop statistics are zero.
+    * Failure: The test will fail if the BGP peer's drop statistics are non-zero.
```

Comment:
Also add that the test will fail if a peer is not configured or if a provided drop stat is not found.